### PR TITLE
Fix "run tests" button and other UI issues

### PR DESCRIPTION
**Pull request by ghengeveld**
- Hides or disables the "run tests" button when running, whatever is appropriate.
- Hides the "Latest snapshot" label when no controls are available, making the whole upper section collapse when there is no need for it.
- Adds a "Code edits detected" message.
- Avoids passing down `startDevBuild` function, instead using a context provider.

<details>
  <summary>📦 Published PR as canary version: <code>1.2.9--canary.199.3e82055.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @chromatic-com/storybook@1.2.9--canary.199.3e82055.0
  # or 
  yarn add @chromatic-com/storybook@1.2.9--canary.199.3e82055.0
  ```
</details>

### CODE REVIEW

File: src/screens/VisualTests/NoBuild.tsx [hunk 1]
```diff
@@ -53,15 +53,16 @@ export const NoBuild = ({
...
 }: NoBuildProps) => {
   const { setAccessToken } = useAuthState();
+  const { isRunning, startBuild, stopBuild } = useRunBuildState();
+
   const getDetails = () => {
     const button = (
-      <Button size="medium" variant="solid" onClick={startDevBuild}>
-        <PlayIcon />
+      <Button size="medium" variant="solid" onClick={isRunning ? stopBuild : startBuild}>
+        {isRunning ? <StopAltIcon /> : <PlayIcon />}
```
Comment (andrewortwein) [inline src/screens/VisualTests/NoBuild.tsx:65]: Do we have any stories that cover this?

File: src/screens/VisualTests/SnapshotControls.tsx [hunk 2]
```diff
@@ -116,68 +112,65 @@ export const SnapshotControls = ({
...
     <>
-      <Label>
-        <Text>
-          <b>
-            {baselineImageVisible ? "Baseline" : "Latest"}
-            <span> snapshot</span>
-          </b>
-        </Text>
-      </Label>
+      {hasControls && (
```
Comment (andrewortwein) [inline src/screens/VisualTests/SnapshotControls.tsx:115]: Do we cover this in a story anywhere?

File: src/screens/VisualTests/SnapshotControls.tsx [hunk 3]
```diff
@@ -330,7 +323,8 @@ export const SnapshotControls = ({
             <ActionButton
               containsIcon
               aria-label={isOutdated ? "Run new tests" : "Rerun tests"}
-              onClick={() => startDevBuild()}
+              onClick={startBuild}
+              disabled={isRunning}
```
Comment (andrewortwein) [inline src/screens/VisualTests/SnapshotControls.tsx:327]: This might be too picky, but it would be nice to also capture this in a story

**Review by andrewortwein — DISMISSED** [review]
Changes look good, but I think we should add a few stories to capture the improvements you made. Please let me know if they're already there and I just missed them.

File: src/screens/VisualTests/NoBuild.tsx (hunk 1 above)
Comment (ghengeveld) [reply src/screens/VisualTests/NoBuild.tsx:65]: Turns out this state isn't actually reachable, as the button won't render at all while the build is running.

File: src/screens/VisualTests/SnapshotControls.tsx (hunk 2 above)
Comment (ghengeveld) [reply src/screens/VisualTests/SnapshotControls.tsx:115]: Yeah all the stories where there is no baseline, there are no controls (and no latest/baseline label).

File: src/screens/VisualTests/SnapshotControls.tsx (hunk 3 above)
Comment (ghengeveld) [reply src/screens/VisualTests/SnapshotControls.tsx:327]: It is, in [Pending Local Build Starting](https://www.chromatic.com/component?appId=6480e1b0042842f149cfd74c&csfId=screens-visualtests-visualtests--pending-local-build-starting&buildNumber=1829&k=65e893ff088ac5e6bef59916-Bottom+2%252Dup-interactive-true&h=15&b=-8) for example.

**Review by thafryer — APPROVED** [review]
LGTM.

**Pull request merged**